--- a/release/src/router/Makefile
+++ b/release/src/router/Makefile
@@ -3660,10 +3660,10 @@ ipset-7.6/configure:

 ipset-7.6/Makefile: ipset-7.6/configure
 	cd ipset-7.6 && \
-	libmnl_CFLAGS="-I$(STAGEDIR)/usr/include" \
-	libmnl_LIBS="-L$(STAGEDIR)/usr/lib -lmnl" \
-	CFLAGS="-Os -Wall $(EXTRACFLAGS) -ffunction-sections -fdata-sections" \
-	LDFLAGS="-Wl,--gc-sections" \
+	libmnl_CFLAGS="-I$(TOP)/libmnl-1.0.4/include" \
+	libmnl_LIBS="-L$(TOP)/libmnl-1.0.4/src/.libs -lmnl" \
+	CFLAGS="-Os -Wall $(EXTRACFLAGS) -ffunction-sections -fdata-sections -I$(TOP)/libmnl-1.0.4/include" \
+	LDFLAGS="-Wl,--gc-sections -L$(TOP)/libmnl-1.0.4/src/.libs -Wl,-rpath-link=$(TOP)/libmnl-1.0.4/src/.libs" \
 	$(CONFIGURE) --prefix=/usr --disable-rpath \
 		--with-kmod=no --with-kbuild=$(LINUXDIR)

